lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -227,13 +341,63 @@ bool BootloaderCom::process()
             m_waitingForResponse = false;
             if(m_cmdProvider->next(m_currentCommand, m_currentResponse))
             {
-                if(true == m_flashManager.sendCommand(
-                    m_currentCommand->command,
-                    m_currentCommand->commandsize))
+                if(m_currentCommand->command == Zumo32U4Specification::SET_MEMORY_ADDR)

```

Comment:
The block from here to line 392 belongs into the cmd provider for the flash process not here.  this function executs cmds from a provider, it does nut build/modify them. That is the duty of the provider not the executer (seperation of concerns).

Reply:
It will be adjusted in the next commit.